plugin-core/src/main/java/appland/rpcService/DefaultAppLandJsonRpcService.java

```diff
@@ -0,0 +1,291 @@
+                })
+                .filter(Objects::nonNull)
+                .forEach(jsonConfigFiles::add);
+
+        try {
+            var payload = new JsonObject();
+            payload.addProperty("jsonrpc", "2.0");
+            payload.addProperty("method", "v1.configuration.set");
+            payload.add("params", GsonUtils.singlePropertyObject("appmapConfigFiles", jsonConfigFiles));
+
+            var json = GsonUtils.GSON.toJson(payload);
+            if (LOG.isTraceEnabled()) {
+                LOG.trace("Sending JSON-RPC message: " + json);
+            }
+
+            HttpRequests.post(serverUrl, json);
+        } finally {
+            project.getMessageBus().syncPublisher(AppLandJsonRpcListener.TOPIC).serverConfigurationUpdated(configFiles);
+        }
+    }
+
+    @Data
+    protected static class JsonRpcServer {
+        @NotNull KillableProcessHandler processHandler;
+        @Nullable Integer jsonRpcPort = null;
+    }
+
+    private class JsonRpcProcessListener extends ProcessAdapter {
+        private final JsonRpcServer jsonRpcServer;
+
+        public JsonRpcProcessListener(JsonRpcServer jsonRpcServer) {
+            this.jsonRpcServer = jsonRpcServer;
+        }
+
+        @Override
+        public void onTextAvailable(@NotNull ProcessEvent event, @NotNull Key outputType) {
+            if (outputType != ProcessOutputTypes.SYSTEM && LOG.isDebugEnabled()) {
+                LOG.debug("JSON-RPC: " + event.getText());
+            }
+
+            var match = PORT_PATTERN.matcher(event.getText().trim());
+            if (match.matches()) {
+                var port = StringUtil.parseInt(match.group(1), -1);
+                if (port > 0) {
+                    jsonRpcServer.jsonRpcPort = port;
+                    triggerSendConfigurationSet();
+                }
+            }
+        }
+
+        @Override
+        public void processTerminated(@NotNull ProcessEvent event) {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("AppMap JSON-RPC server terminated: " + event);
+            }
+
+            synchronized (DefaultAppLandJsonRpcService.this) {
+                DefaultAppLandJsonRpcService.this.jsonRpcServer = null;
+                jsonRpcServer.jsonRpcPort = null;
+            }
+
+            var currentRestartDelay = nextRestartDelay;
+            var restartNeeded = currentRestartDelay >= 0L && currentRestartDelay <= MAX_RESTART_DELAY_MILLIS;
+            if (restartNeeded) {
+                nextRestartDelay = (long) ((double) currentRestartDelay * NEXT_RESTART_FACTOR);
+                AppExecutorUtil.getAppScheduledExecutorService().schedule(() -> {
+                            try {
+                                DefaultAppLandJsonRpcService.this.startServer();
+                            } finally {
+                                project.getMessageBus().syncPublisher(AppLandJsonRpcListener.TOPIC).serverRestarted();
+                            }
+                        },
```

Comment:
Similarly if `APPMAP_API_KEY` changes the service would need to be restarted with the new value